Robot Code/src/org/usfirst/frc/team3316/robot/commands/hood/HoodPID.java

```diff
@@ -0,0 +1,76 @@
+package org.usfirst.frc.team3316.robot.commands.hood;
+
+import org.usfirst.frc.team3316.robot.Robot;
+import org.usfirst.frc.team3316.robot.commands.DBugCommand;
+
+import edu.wpi.first.wpilibj.PIDController;
+import edu.wpi.first.wpilibj.PIDOutput;
+import edu.wpi.first.wpilibj.PIDSource;
+import edu.wpi.first.wpilibj.PIDSourceType;
+
+public class HoodPID extends DBugCommand
+{
+	private PIDController pid;
+	private double pidOutput;
+
+	public HoodPID() {
+		
+		requires(Robot.hood);
+		
+		pid = new PIDController(0, 0, 0, new PIDSource()
+		{
+			public void setPIDSourceType(PIDSourceType pidSource)
+			{
+			}
+			public double pidGet()
+			{
+				return Robot.hood.getAngle();
+			}
+
+			public PIDSourceType getPIDSourceType()
+			{
+				return PIDSourceType.kDisplacement;
+			}
+		}, new PIDOutput()
+		{
+			public void pidWrite(double output)
+			{
+				pidOutput = output;
+			}
+		});
+		
+		pid.setOutputRange(-1, 1);
+	}
+
+	protected void init()
+	{
+		pid.enable();
+	}
+
+	protected void execute()
+	{
+		pid.setPID((double) config.get("hood_Pid_Kp"),
+					(double) config.get("hood_Pid_Ki"),
+					(double) config.get("hood_Pid_Kd"));
+		
+		pid.setSetpoint((double) config.get("hood_Pid_Angle"));
```

Comment:
How are we gonna integrate into this? We'll run a command which modify the config? or perhaps create an overwriting command which on execute gets the reading off the camera and adjusts the setpoint?